### PR TITLE
Fix explicit DEFAULT value in insert query

### PR DESCRIPTION
**Pull request by Wuvist**
It seems that `go-mysql-server` is not compatible with insert query like:

```sql
INSERT INTO `users` (`id`, `deleted`) VALUES (1, DEFAULT)
```

Whereas `DEFAULT` is just to specify `deleted` columns should use column [default value explicitly](https://dev.mysql.com/doc/refman/8.0/en/insert.html).

The issue could be demostrated using below code:
```go
package main

import (
	dbsql "database/sql"

	sqle "github.com/dolthub/go-mysql-server"
	"github.com/dolthub/go-mysql-server/auth"
	"github.com/dolthub/go-mysql-server/memory"
	"github.com/dolthub/go-mysql-server/server"
	_ "github.com/go-sql-driver/mysql"
)

func main() {
	db, _ := dbsql.Open("mysql", "root:@tcp(127.0.0.1:3307)/test")
	defer db.Close()

	query := `CREATE TABLE users (
  id bigint(20) unsigned NOT NULL AUTO_INCREMENT,
  deleted tinyint(1) unsigned NOT NULL DEFAULT '0',
  PRIMARY KEY (id)
) ENGINE=InnoDB AUTO_INCREMENT=1 DEFAULT CHARSET=utf8mb4`

	stmt, _ := db.Prepare(query)
	stmt.Exec()

	_, err := db.Exec("INSERT INTO `users` (`id`, `deleted`) VALUES (1, DEFAULT)")
	if err != nil {
		panic(err.Error())
	}

	stmtOut, _ := db.Prepare("SELECT `deleted` FROM `users` WHERE `id` = 1")
	defer stmtOut.Close()

	deleted := true
	err = stmtOut.QueryRow().Scan(&deleted)
	if err != nil {
		panic(err.Error())
	}

	if deleted == true {
		panic("Wrong deleted value")
	}
}

var engine *sqle.Engine

func init() {
	engine = sqle.NewDefault()
	db := memory.NewDatabase("test")
	engine.AddDatabase(db)

	config := server.Config{
		Protocol: "tcp",
		Address:  "localhost:3307",
		Auth:     auth.NewNativeSingle("root", "", auth.AllPermissions),
	}

	s, _ := server.NewDefaultServer(config, engine)

	go s.Start()
}
```

```
`db.Exec("INSERT INTO `users` (`id`, `deleted`) VALUES (1, DEFAULT)")` will result in error:

Error 1105: plan is not resolved because of node '*plan.Values'
```

This pull request should avoid such issue by turning explicit `DEFAULT` in insert query to implicit.

### CODE REVIEW

File: sql/parse/parse.go [hunk 1]
```diff
@@ -1452,7 +1452,32 @@ func convertInsert(ctx *sql.Context, i *sqlparser.Insert) (sql.Node, error) {
 		ignore = true
 	}
 
-	return plan.NewInsertInto(sql.UnresolvedDatabase(i.Table.Qualifier.String()), tableNameToUnresolvedTable(i.Table), src, isReplace, columnsToStrings(i.Columns), onDupExprs, ignore), nil
+	columnWithDefaultValues := make(map[int]bool)
+
+	if e, ok := src.(*plan.Values); ok {
+		var needCols []sql.Expression
+		for i, s := range e.ExpressionTuples[0] {
```
Comment (zachmu) [inline sql/parse/parse.go:1459]: There can be more than 1 element in ExpressionTuples, need to account for that

**Review by zachmu — COMMENTED** [review]
Thanks for the contribution!

This looks good, but please add tests:

1) parse_test.go
2) insert_queries.go

And don't forget to run ./format_repo.sh when you're done.

**Comment by Wuvist** [comment]
@zachmu Updated pull request with tests and handles multiple ExpressionTuples.

However, as my method is just to turn explicit DEFAULT to implicit. So, it won't work for insertion query test case like this:
```go
	{
		Name: "explicit DEFAULT with multiple values different order",
		SetUpScript: []string{
			"CREATE TABLE mytable(id int PRIMARY KEY, v2 int NOT NULL DEFAULT '2', v3 int NOT NULL DEFAULT '3')",
		},
		Assertions: []ScriptTestAssertion{
			{
				Query: "INSERT INTO mytable (id, v2, v3)values (1, DEFAULT, 0), (2, 0, DEFAULT)",
				Expected: []sql.Row{
					{sql.OkResult{RowsAffected: 2}},
				},
			},
			{
				Query: "SELECT * FROM mytable",
				Expected: []sql.Row{
					{1, 2, 0},
					{2, 0, 3},
				},
			},
		},
	},
```

**Review by zachmu — APPROVED** [review]
LGTM, thanks for the changes!

**Comment by zachmu** [comment]
https://github.com/dolthub/go-mysql-server/issues/527

**Pull request merged**